controllers/helpers_test.go

```diff
@@ -0,0 +1,855 @@
+				Values: aws.StringSlice([]string{"vpc-exists"}),
+			},
+		}})).Return(&ec2.DescribeSubnetsOutput{
+		Subnets: []*ec2.Subnet{
+			{
+				VpcId:               aws.String("vpc-exists"),
+				SubnetId:            aws.String("subnet-1"),
+				AvailabilityZone:    aws.String("us-east-1a"),
+				CidrBlock:           aws.String("10.0.10.0/24"),
+				MapPublicIpOnLaunch: aws.Bool(false),
+			},
+			{
+				VpcId:               aws.String("vpc-exists"),
+				SubnetId:            aws.String("subnet-2"),
+				AvailabilityZone:    aws.String("us-east-1c"),
+				CidrBlock:           aws.String("10.0.11.0/24"),
+				MapPublicIpOnLaunch: aws.Bool(false),
+				Tags: []*ec2.Tag{
+					{
+						Key:   aws.String("sigs.k8s.io/cluster-api-provider-aws/cluster/test-cluster"),
+						Value: aws.String("owned"),
+					},
+					{
+						Key:   aws.String("sigs.k8s.io/cluster-api-provider-aws/role"),
+						Value: aws.String("public"),
+					},
+					{
+						Key:   aws.String("Name"),
+						Value: aws.String("test-cluster-subnet-public"),
+					},
+					{
+						Key:   aws.String("kubernetes.io/cluster/test-cluster"),
+						Value: aws.String("shared"),
+					},
+				},
+			},
+		},
+	}, nil)
+	m.DescribeRouteTables(gomock.Eq(&ec2.DescribeRouteTablesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: aws.StringSlice([]string{"vpc-exists"}),
+			},
+		}})).Return(&ec2.DescribeRouteTablesOutput{
+		RouteTables: []*ec2.RouteTable{
+			{
+				Routes: []*ec2.Route{
+					{
+						GatewayId: aws.String("igw-12345"),
+					},
+				},
+			},
+		},
+	}, nil)
+	m.DescribeNatGatewaysPages(gomock.Eq(&ec2.DescribeNatGatewaysInput{
+		Filter: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{aws.String("vpc-exists")},
+			},
+			{
+				Name:   aws.String("state"),
+				Values: aws.StringSlice([]string{ec2.VpcStatePending, ec2.VpcStateAvailable}),
+			},
+		}}), gomock.Any()).Return(nil)
+	m.DescribeVpcs(gomock.Eq(&ec2.DescribeVpcsInput{
+		VpcIds: []*string{
+			aws.String("vpc-exists"),
+		},
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("state"),
+				Values: aws.StringSlice([]string{ec2.VpcStatePending, ec2.VpcStateAvailable}),
+			},
+		},
+	})).
+		Return(&ec2.DescribeVpcsOutput{
+			Vpcs: []*ec2.Vpc{
+				{
+					State:     aws.String("available"),
+					VpcId:     aws.String("vpc-exists"),
+					CidrBlock: aws.String("10.0.0.0/8"),
+					Tags: []*ec2.Tag{
+						{
+							Key:   aws.String("sigs.k8s.io/cluster-api-provider-aws/role"),
+							Value: aws.String("common"),
+						},
+						{
+							Key:   aws.String("Name"),
+							Value: aws.String("test-cluster"),
+						},
+					},
+				},
+			},
+		}, nil)
+}
+
+func mockedCreateMaximumVPCCalls(m *mock_ec2iface.MockEC2APIMockRecorder) {
+	m.CreateVpc(gomock.AssignableToTypeOf(&ec2.CreateVpcInput{})).Return(nil, errors.New("The maximum number of VPCs has been reached"))
+}
+
+func mockedDeleteVPCCalls(m *mock_ec2iface.MockEC2APIMockRecorder) {
```

Comment:
These calls probably won't be shared with other tests? If so, could go in controller file.

Reply:
I think subset of these calls might be common in machine controller integration tests as well, that's why moved for now, but can revert it and make changes while raising PR for machine controller integration tests, wdyt?

Reply:
Done